packages/opentelemetry-sdk-trace-base/src/export/SimpleSpanProcessor.ts

```diff
@@ -35,18 +35,28 @@ import { SpanExporter } from './SpanExporter';
  */
 export class SimpleSpanProcessor implements SpanProcessor {
   private _shutdownOnce: BindOnceFuture<void>;
+  private _unresolvedResources: Set<Promise<void>>;
 
   constructor(private readonly _exporter: SpanExporter) {
     this._shutdownOnce = new BindOnceFuture(this._shutdown, this);
+    this._unresolvedResources = new Set<Promise<void>>();
   }
 
-  forceFlush(): Promise<void> {
-    // do nothing as all spans are being exported without waiting
+  async forceFlush(): Promise<void> {
+    // await unresolved resources before resolving
+    for (const unresolvedResource of this._unresolvedResources) {
+      await unresolvedResource;
+    }
+
     return Promise.resolve();
   }
 
-  // does nothing.
-  onStart(_span: Span, _parentContext: Context): void {}
+  onStart(_span: Span, _parentContext: Context): void {
+    // store the resource's unresolved promise
+    if (!_span.resource.asyncAttributesHaveResolved()) {
+      this._unresolvedResources.add(_span.resource.waitForAsyncAttributes());
+    }
```

Comment:
I think it might make more sense to do it in `onEnd` only if we actually delay the export

Reply:
So the idea was to store the promise as soon as the span starts since the `forceflush` could (correct me if I'm wrong) be called before getting to the `onEnd` and thus not exporting correctly